### PR TITLE
minor IMAP string handling fixes

### PR DESCRIPTION
**Pull request by alyptik**
While looking for the cause of my IMAP truncation bug I noticed a few minor places that needed minor fixes along the way (mostly snprintf related) that were a little sparse on error checking and other small things like nesting confusing expressions a bit too deeply and incorrect return types.

The commits are all fairly small and simple to read over except for maybe:

* commit [f8385f5](https://github.com/neomutt/neomutt/commit/f8385f5bbd22d563a5390183b45ff9147994452b) (message.c: replace nested conditional operators with static string table)

which I couldn't really figure out any good ways to simplify without drastically affecting semantics.

### CODE REVIEW

File: bcache.c [hunk 1]
```diff
@@ -70,7 +70,7 @@ static int bcache_path(struct Account *account, const char *mailbox, char *dst,
   }
 
   size_t mailboxlen = mutt_str_strlen(mailbox);
-  len = snprintf(dst, dstlen - 1, "%s/%s%s%s", MessageCachedir, host, NONULL(mailbox),
```
Comment (gahr) [inline bcache.c]: Using `dstlen` is correct. The only caller uses `sizeof(bcache->path)`, so this argument is really the number of bytes available in the buffer, not the length of the longest C string that would fit in there.

File: imap/message.c [hunk 2]
```diff
@@ -693,7 +694,11 @@ int imap_read_headers(struct ImapData *idata, unsigned int msn_begin, unsigned i
     mutt_progress_init(&progress, _("Evaluating cache..."), MUTT_PROGRESS_MSG,
                        ReadInc, msn_end);
 
-    snprintf(buf, sizeof(buf), "UID FETCH 1:%u (UID FLAGS)", uidnext - 1);
+    if (snprintf(buf, sizeof(buf), "UID FETCH 1:%u (UID FLAGS)", uidnext - 1) < 0)
```
Comment (gahr) [inline imap/message.c]: Is there any reasonable situation in which this snprintf could fail, that you can think of?

File: imap/message.c [hunk 3]
```diff
@@ -1051,10 +1061,21 @@ int imap_fetch_message(struct Context *ctx, struct Message *msg, int msgno)
    * command handler */
   h->active = false;
 
-  snprintf(buf, sizeof(buf), "UID FETCH %u %s", HEADER_DATA(h)->uid,
-           (mutt_bit_isset(idata->capabilities, IMAP4REV1) ?
-                (ImapPeek ? "BODY.PEEK[]" : "BODY[]") :
-                "RFC822"));
+  /* the IMAP4REV1 predicate being true will always result in one of our two
```
Comment (gahr) [inline imap/message.c]: Sorry, but I find the original code clearer.

File: imap/message.c (hunk 2 above)
Comment (alyptik) [reply imap/message.c]: That's a good point, I'll revert this one.

File: imap/message.c (hunk 3 above)
Comment (alyptik) [reply imap/message.c]: Figured this might be a bit too weird; will revert this as well.

**Comment by alyptik** [comment]
Dropped the requested commits.

Although I wasn't sure if the comment on:

* de2bc27 (bcache.c: fix off-by-one sprintf() dest buffer size error)

was requesting any changes?

**Comment by gahr** [comment]
Yes it was :) the code is correct as it is. I would also ask you to revert the reindentation of the header file. Thank you!

**Comment by alyptik** [comment]
No problem; header change has been reverted.

**Comment by alyptik** [comment]
Whoops, I reverted the prototype as well by accident, fixed.

**Comment by gahr** [comment]
@alyptik sorry for the confusion about your change to `bcache_path`. I read the diff the other way around: your change is correct.

**Comment by alyptik** [comment]
No worries, cheers.

**Pull request merged**